looper/looper.py

```diff
@@ -877,35 +877,54 @@ def uniqify(seq):
 
 
 
-def check(prj):
+def fetch_flag_files(prj):
     """
-    Check Project status, based on flag files.
-    
-    :param Project prj: Project for which to inquire about status
+    Find all flag file paths for the given project.
+
+    :param Project | AttributeDict prj: full Project or AttributeDict with
+        similar metadata and access/usage pattern
+    :return Mapping[str, list[str]]: collection of filepaths associated with
+        particular flag for samples within the given project
     """
 
-    # TODO: resume here to hook into specific protocols, and if we want
-    # TODO (continued): to just use Python rather than shelling out.
-    from collections import defaultdict
-    flags_by_protocol = defaultdict(list)
+    def _glob_expr(flag):
+        return os.path.join(
+            prj.metadata.results_subdir, "*", "{}.flag".format(flag))
+
+    # TODO: import from pep.
+    flags = ["completed", "running", "failed", "waiting", "partial"]
 
-    # prefix
-    pf = "ls " + prj.metadata.results_subdir + "/"
-    cmd = os.path.join(pf + "*/*.flag | xargs -n1 basename | sort | uniq -c")
-    _LOGGER.info(cmd)
-    subprocess.call(cmd, shell=True)
+    return {f: glob.glob(_glob_expr(f)) for f in flags}
 
-    flags = ["completed", "running", "failed", "waiting"]
 
-    counts = {}
-    for f in flags:
-        counts[f] = int(subprocess.check_output(
-                pf + "*/*" + f + ".flag 2> /dev/null | wc -l", shell=True))
 
-    for f, count in counts.items():
-        if 0 < count < 30:
-            _LOGGER.info(f + " (" + str(count) + ")")
-            subprocess.call(pf + "*/*" + f + ".flag 2> /dev/null", shell=True)
+class Checker(Executor):
```

Comment:
is there an advantage in this case to making this a function object instead of just a function?